solr/core/src/java/org/apache/solr/filestore/PackageStoreAPI.java

```diff
@@ -158,6 +159,17 @@ public void upload(SolrQueryRequest req, SolrQueryResponse rsp) {
           List<String> signatures = readSignatures(req, buf);
           MetaData meta = _createJsonMetaData(buf, signatures);
           PackageStore.FileType type = packageStore.getType(path, true);
+          boolean[] returnAfter = new boolean[]{false};
```

Comment:
@noblepaul why is this an array and not just a plain boolean? I don't follow.

Reply:
this is used in a lambda . if you keep it as a boolean, it will not compile